lib/stateView/document/fetchDocumentsFactory.js

```diff
@@ -15,7 +25,29 @@ function fetchDocumentsFactory(createSVDocumentRepository) {
    */
   async function fetchDocuments(contractId, type, options, stateViewTransaction = undefined) {
     const svDocumentRepository = createSVDocumentRepository(contractId, type);
+
+    const svContract = await svContractMongoDbRepository.find(contractId);
+    if (!svContract) {
+      return [];
```

Comment:
I think we need `InvalidDataContractError` such as Type and Query.

Reply:
Missed it(